pkg/controller/controller.go

```diff
@@ -671,3 +681,25 @@ func setObjectKind(obj runtime.Object) {
 	}
 	ok.SetGroupVersionKind(gvk)
 }
+
+type bpCache struct {
+	bpMap map[string]*crv1alpha1.Blueprint
+}
+
+func initBpCache(ctx context.Context, cli versioned.Interface, ns string) (*bpCache, error) {
```

Comment:
I'm wondering why `initBpCache` is a separate function? Did you consider it to be a part of `Controller`?

Reply:
Not, it's assumed to be a part of `onAdd` function context.

Reply:
Fixed